### DISCUSSION TITLE
Notification support

### DISCUSSION THREAD

**Discussion by klochowicz** [get10101/10101#953] [Ideas]
- [Rationale](#sec-1)
    - [Reasons](#sec-1-1)
  - [How to do it](#sec-2)
  - [Technical implementation](#sec-3)
    - [Firebase](#sec-3-1)
    - [Local notifications](#sec-3-2)
  - [Side notes](#sec-4)

# Rationale<a id="sec-1"></a>

-   Notifications are an essential part of any mobile applications

## Reasons<a id="sec-1-1"></a>

-   nudge the user for more activity
    -   the red squiggles (actions) prompt the user to open the app
    -   we could post news about new blog posts, BTC price etc.
-   communicate about downtime, errors etc
-   ask to tweet / fill a survey
-   notify about their payment status / position (e.g. expiry)
-   prompt the user to start the app once a day

# How to do it<a id="sec-2"></a>

I would like to propose adding both Firebase and local notification systems in our app. Firebase app could communicate about errors, pass on surveys etc. We could **also** inform about the availability of new version, prompting the user to upgrade their 10101 wallet.

Local ones are only active when the app is up, but they can tell e.g. that a position got closed, inform about the profit / loss after a position got closed, communicate that a payment has been received etc.

If anything, I believe that firebase notification service should be added first, because:

-   it's easier to implement
-   will give us means to reach out the users to both communicate about downtime and to increase app usage

# Technical implementation<a id="sec-3"></a>

Initial ticket does not need to cover the whole range of notifications. One is enough.

Below are some links to look at:

## Firebase<a id="sec-3-1"></a>

-   <https://stackoverflow.com/questions/72749963/push-notifications-for-flutter-chat-app-without-firebase>
-   <https://medium.com/@yshean/implementing-push-notifications-in-flutter-apps-aef98451e8f1>

## Local notifications<a id="sec-3-2"></a>

-   <https://dev.to/carminezacc/flutter-notifications-without-firebase-33d2>

# Side notes<a id="sec-4"></a>

-   having our tweets / blog posts in the app would be nice.

**Comment by bonomat** [comment]
That's not much of a discussion 😅 Sounds good as you describe it :) 

Let's convert this to a ticket?

**Reply by klochowicz** [reply]
It's discussion because it was unknown to me if and when we'd like the feature :) 

I'll make separate tickets for both firebase and local notifications.

**Reply by bonomat** [reply]
Let's close this discussion then :)
